src/main/java/com/google/devtools/build/lib/hash/Blake3Hasher.java

```diff
@@ -0,0 +1,184 @@
+package com.google.devtools.build.lib.hash;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import com.google.common.hash.Funnel;
+import com.google.common.hash.HashCode;
+import com.google.common.hash.Hasher;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.nio.charset.Charset;
+
+public class Blake3Hasher implements Hasher {
```

Comment:
This isn't used directly by `DigestHashFunction`, so it can be package-private.  It's also not meant to be extended ever, so it can be final.  So, `final class Blake3Hasher` instead of `public class Blake3Hasher`

Reply:
Done.